### PR TITLE
Update webidl2 to v13.0.3

### PR DESCRIPTION
**Pull request by saschanaz**
Replaces and closes #11073.

### CODE REVIEW

**Comment by saschanaz** [comment]
CC'ing @foolip who authored #11073.

File: resources/test/tests/unit/basic.html [hunk 1]
```diff
@@ -24,7 +24,12 @@
         assert_equals(typeof WebIDL2.parse, "function");
     }, 'WebIDL2 namespace should have a parse method');
     test(function() {
-        assert_throws(new TypeError(), function() {  WebIDL2.parse("I'm a syntax error"); });
```
Comment (gsnedders) [inline resources/test/tests/unit/basic.html]: Why rewrite this to not use `assert_throws`?

Comment (gsnedders) [reply resources/test/tests/unit/basic.html]: (Like, yes, the error type has changed, but can we not still use `assert_throws`?)

**Comment by csnardi** [comment]
A couple test changes are needed due to parsing becoming stricter:

- `fetch/api/idl.any.js` cannot reference an empty enum, as that is now invalid. #9847 would pull in that IDL file, which could then be used as a dependency IDL.
- `webaudio/the-audio-api/the-audiobuffer-interface/idl-test.html` and `webaudio/the-audio-api/the-audiodestinationnode-interface/idl-test.html` do not escape angle brackets, creating text that is now rejected as invalid. Really, these tests should be deleted in favor of `webaudio/idlharness.https.html`, but per https://wpt.fyi/results/webaudio/idlharness.https.html Chrome is the only browser that actually can run the test.
- `pointerevents/extension/idlharness.html` also does not escape angle brackets, which could be fixed by moving the IDL to `interfaces/` (but what should it be named? It's an extension of the pointer events spec).

**Comment by foolip** [comment]
>  `pointerevents/extension/idlharness.html` also does not escape angle brackets, which could be fixed by moving the IDL to `interfaces/` (but what should it be named? It's an extension of the pointer events spec).

Spec is https://w3c.github.io/pointerevents/extension.html so I suggest `pointerevents-extension.idl`.

File: resources/test/tests/unit/basic.html (hunk 1 above)
Comment (saschanaz) [reply resources/test/tests/unit/basic.html]: `assert_throws` seems to allow only DOMException and JS native error objects that have `name` property. Should the function be fixed to allow arbitrary error objects?

**Comment by saschanaz** [comment]
@csnardi 

>creating text that is now rejected as invalid

Should they just be escaped before deletion?

**Comment by csnardi** [comment]
@saschanaz Probably, I'm not sure a good way to make `webaudio/idlharness.https.html` run without failing immediately on the other browsers, so that seems like the only way to do it for now without losing test coverage.

**Comment by foolip** [comment]
Please let me know if you'd like me to try out the change in Chromium's CQ, since that can help find breakage not found by our Travis setup here.

**Comment by saschanaz** [comment]
@foolip That would be a great help! 👍

**Comment by csnardi** [comment]
Since the last run didn't pick up my change to `fetch/api/idl.any.js`, I re-ran the failing jobs and now all tests are passing.

**Comment by foolip** [comment]
Excellent, thanks @csnardi!

**Comment by gsnedders** [comment]
Do they somehow pass despite https://github.com/web-platform-tests/wpt/pull/11496#issuecomment-397358030? does that mean we can land this?

**Comment by csnardi** [comment]
The last two items were fixed inside of this PR (and also included in the try run). I think this could be landed, assuming we can't simplify the `assert_throws` change.

**Review by gsnedders — APPROVED** [review]

File: resources/test/tests/unit/basic.html (hunk 1 above)
Comment (gsnedders) [reply resources/test/tests/unit/basic.html]: Made https://github.com/web-platform-tests/wpt/issues/11556 for this

**Pull request merged**